libc/src/stdlib/rand.cpp

```diff
@@ -13,20 +13,33 @@
 
 namespace LIBC_NAMESPACE {
 
-// An implementation of the xorshift64star pseudo random number generator. This
+// An implementation of the xorshift* pseudo random number generator. This
 // is a good general purpose generator for most non-cryptographics applications.
-LLVM_LIBC_FUNCTION(int, rand, (void)) {
+static inline unsigned long xorshiftstar(unsigned long a, unsigned long b,
```

Comment:
It's more code, but I think it's much clearer if we just call this `xorshift64*` and `xorshift32` and have them be separate functions.

Reply:
This request introduces a warning when the other function is not selected:
```
/home/mgadelha/tools/llvm-project/libc/src/stdlib/rand.cpp:19:12: warning: function 'xorshift64star' is not needed and will not be emitted [-Wunneeded-internal-declaration]
   19 | static int xorshift64star() {
      |            ^~~~~~~~~~~~~~
1 warning generated.